ocs_ci/ocs/ocp.py

```diff
@@ -307,10 +307,15 @@ def wait_for_resource(
                 in_condition = []
                 sample = sample['items']
                 for item in sample:
-                    if self.get_resource_status(
-                        item.get('metadata').get('name')
-                    ) == condition:
-                        in_condition.append(item)
+                    try:
+                        _name = item.get('metadata').get('name')
+                        if self.get_resource_status(_name) == condition:
+                            in_condition.append(item)
+                    except CommandFailed as ex:
+                        log.info(
+                            f"Failed to get status of resource: {_name}, "
```

Comment:
a) we cannot handle this in get_resource_status as here: 
https://github.com/red-hat-storage/ocs-ci/blob/107e72966781f0ccffe4eec1e33b6015884e8d3b/ocs_ci/ocs/ocp.py#L303 
We don't want to have this be handled as some ppl can expect this exception in their tests. See also some conversation about this here: https://github.com/red-hat-storage/ocs-ci/pull/640/files#diff-0acc23436a82754145dad2f2c3e0536dR312-R320
b)This happen only when the resource get deleted during the loop otherwise we shouldn't get here to this status I think. And than it's ok to write Failed to get status as the resource got deleted.